wai-extra/Network/Wai/Parse.hs

```diff
@@ -796,17 +796,35 @@ sinkTillBound bound iter seed0 src max' = do
     b <- final
     return (b, seed)
 
-parseAttrs :: S.ByteString -> [(S.ByteString, S.ByteString)]
-parseAttrs = map go . S.split _semicolon
-  where
-    tw = S.dropWhile (== _space)
-    dq s =
-        if S.length s > 2 && S.head s == _quotedbl && S.last s == _quotedbl
-            then S.tail $ S.init s
-            else s
-    go s =
-        let (x, y) = breakDiscard _equal s
-         in (tw x, dq $ tw y)
+parseContentDispositionAttrs :: S.ByteString -> [(S.ByteString, S.ByteString)]
+parseContentDispositionAttrs = parseTokenValues
+ where
+    nonTokenChars = [_semicolon, _equal]
+    dropSpace = S.dropWhile (== _space)
+    parseTokenValues input | S.null input = []
+    parseTokenValues input =
+        let (token, rest) = parseToken $ dropSpace input
+         in case S.uncons rest of
+            Just (c, rest')
+                | c == _equal -> 
+                    let (value, rest'') = parseValue $ dropSpace rest'
```

Comment:
The `dropSpace` here is superfluous, since `parseValue` will also do `dropSpace`.